docs/spec_draft.md

```diff
@@ -343,6 +344,120 @@ logical operation.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.batch_norm_grad
+
+### Semantics
+
+Computes gradients of `operand`, `scale` and offset, using `mean`, `variance`
+and `grad_output`, the gradient of output from `batch_norm_training`, and
+produces `grad_operand`, `grad_scale` and `grad_offset` tensors. More formally,
+this operation can be expressed as a decomposition to existing StableHLO
+operations using Python-like syntax as follows:
+
+```python
+def reduce_sum(operand, feature_index):
+  (sum,) = reduce(
+      inputs=[operand],
+      init_values=[0],
+      dimensions=[i for i in range(rank(operand)) if i != feature_index],
+      body=lambda x, y: add(x, y))
+  return sum
+
+def compute_mean(operand, feature_index):
+  sum = reduce_sum(operand, feature_index)
+  divisor = constant(num_elements(operand) / dim(operand, feature_index))
```

Comment:
drop constant?

Reply:
LGTM except for the use of `constant`. The StabbleHLO constant has a implicit shape (provided as output of the op) which is not evident here